EncoreTickets.SDK.Tests/UnitTests/Aws/AwsSqsTests.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System.Threading;
+using System.Threading.Tasks;
+using Amazon.Extensions.NETCore.Setup;
+using Amazon.Runtime.CredentialManagement;
+using Amazon.SQS;
+using Amazon.SQS.Model;
+using EncoreTickets.SDK.Aws;
+using Moq;
+using NUnit.Framework;
+
+namespace EncoreTickets.SDK.Tests.UnitTests.Aws
+{
+    internal class AwsSqsTests : AwsSqs
+    {
+        public static AwsSqsMockers Mockers;
+
+        static AwsSqsTests()
+        {
+            Mockers = new AwsSqsMockers();
+        }
+
+        public AwsSqsTests() : base("profile_name", "region_name")
+        {
+        }
+
+        protected override ICredentialProfileStore CreateCredentialProfileStore()
+        {
+            return Mockers.SharedCredentialsFileMock.Object;
+        }
+
+        protected override IAmazonSQS CreateAmazonSqsClient(AWSOptions options)
+        {
+            return Mockers.ClientMock.Object;
+        }
+
+        [TestCase("dev", "eu-west-1")]
+        public void Constructor_IfProfileAndRegionArguments_InitializesClient(string profileName, string regionName)
+        {
+            AwsSqsWrapper.Mockers = new AwsSqsMockers();
```

Comment:
When you assign to a static field in a unit test, you are introducing a dependency on the order of execution of the tests unless you are very careful. Are you sure you need this?

Reply:
We need this property as static, because it must be initialized before calling the AwsSqs constructor. What would you suggest to replace this?

Reply:
Can't we initialize it in AwsSqsWrapper itself?

Reply:
We can do this, but the created static method must be called before the constructor

Reply:
But aren't static members initialized before first call to the constructor anyway?

Reply:
Do you mean static constructor? But if we do not reinitialize the mocker object before each test that uses this, we may get an error in the number of calls to mocked methods

Reply:
Oh, I see... But why can't we create this object here and then pass it in the constructor along with everything else? This would create it before the base constructor is called and allow us to make Mockers instance property instead of static so it should be possible to still run tests in parallel

Reply:
Yes, it will be created before the constructor is called, but assigned after the call to the base constructor, in which the mocked method is used. First, the base constructor will be called, and after that the AwsSqsWrapper constructor

Reply:
Is it possible to somehow refactor the AwsSqs class then so that we don't have to create some weird protected methods that only exist so that we could verify they are being called? Like moving the logic in question to a separate class perhaps?